RF_PBMC.R

```diff
@@ -0,0 +1,119 @@
+	seuratObj <- NormalizeData(object = seuratObj, normalization.method = "LogNormalize", scale.factor = 10000)
+	seuratObj <- FindVariableGenes(object = seuratObj, mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, x.high.cutoff = 3, y.cutoff = 0.5)
+	length(x = seuratObj@var.genes)
+	seuratObj <- ScaleData(object = seuratObj, vars.to.regress = c("nUMI", "percent.mito"))
+	seuratObj <- RunPCA(object = seuratObj, pc.genes = seuratObj@var.genes, do.print = TRUE, pcs.print = 1:5, genes.print = 5)
+	PrintPCA(object = seuratObj, pcs.print = 1:5, genes.print = 5, use.full = FALSE)
+	viz <- VizPCA(object = seuratObj, pcs.use = 1:2)
+	print(viz)
+	pca <- PCAPlot(object = seuratObj, dim.1 = 1, dim.2 = 2)
+	seuratObj <- ProjectPCA(object = seuratObj, do.print = FALSE)
+	seuratObj <- JackStraw(object = seuratObj, num.replicate = 100, display.progress = FALSE)
+	JackStrawPlot(object = seuratObj, PCs = 1:12)
+	jackstraw <- JackStrawPlot(object = seuratObj, PCs = 1:12)
+	print(jackstraw)
+	elbow <- PCElbowPlot(object = seuratObj)
+	print(elbow)
+	seuratObj <- FindClusters(object = seuratObj, reduction.type = "pca", dims.use = 1:10, resolution = 0.6, print.output = 0, save.SNN = TRUE)
+	PrintFindClustersParams(object = seuratObj)
+	seuratObj <- RunTSNE(object = seuratObj, dims.use = 1:10, do.fast = TRUE)
+	TSNEPlot(object = seuratObj)
+	# find all markers of cluster 1
+	cluster1.markers <- FindMarkers(object = seuratObj, ident.1 = 1, min.pct = 0.25)
+	print(x = head(x = cluster1.markers, n = 5))
+	# find all markers distinguishing cluster 5 from clusters 0 and 3
+	cluster5.markers <- FindMarkers(object = seuratObj, ident.1 = 5, ident.2 = c(0, 3), min.pct = 0.25)
+	print(x = head(x = cluster5.markers, n = 5))
+	# find markers for every cluster compared to all remaining cells, report
+	# only the positive ones
+	seuratObj.markers <- FindAllMarkers(object = seuratObj, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
+	seuratObj.markers %>% group_by(cluster) %>% top_n(2, avg_logFC)
+	cluster1.markers <- FindMarkers(object = seuratObj, ident.1 = 0, thresh.use = 0.25, test.use = "roc", only.pos = TRUE)
+	top10 <- seuratObj.markers %>% group_by(cluster) %>% top_n(10, avg_logFC)
+	return(seuratObj)
+}
+
+# generate random forest classifier 
+# save in tsv file - gene names for top 100 factors from the classification model in order of importance
+classify_data <- function(seuratObj){
+	classifier_input_x = t(seuratObj@data)
+	print(classifier_input_x)
+	classifier_input_y = as.factor(as.numeric(seuratObj@ident))
+	trained_classifier = randomForest(x = as.matrix(classifier_input_x) , y=factor(classifier_input_y) , importance = TRUE )
+	print(trained_classifier) 
+	ranked_list = trained_classifier$importance[order(-trained_classifier$importance[,'MeanDecreaseAccuracy']),]
+	dput(rownames(ranked_list)[1:300]) 
+	FINAL_LIST = rownames(ranked_list)[1:100]
```

Comment:
line 73 and 74 should use optional parameters for the subsetting ranges.

Reply:
Comment added indicating TODO for parameterization